lib/ui/tab-navigation.tsx

```diff
@@ -0,0 +1,144 @@
+import { cn } from "@/lib/utils"
+import * as NavigationMenuPrimitives from "@radix-ui/react-navigation-menu"
+import { motion } from "framer-motion"
+import * as React from "react"
+
+function getSubtree(
+  options: { asChild: boolean | undefined; children: React.ReactNode },
+  content: React.ReactNode | ((children: React.ReactNode) => React.ReactNode)
+) {
+  const { asChild, children } = options
+  if (!asChild)
+    return typeof content === "function" ? content(children) : content
+
+  const firstChild = React.Children.only(children) as React.ReactElement
+  return React.cloneElement(firstChild, {
+    children:
+      typeof content === "function"
+        ? content(firstChild.props.children)
+        : content,
+  })
+}
+
+interface TabNavigationProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof NavigationMenuPrimitives.Root>,
+    "orientation" | "defaultValue" | "dir"
+  > {
+  type?: "primary" | "secondary"
```

Comment:
what do you think if we have a `secondary` boolean prop and that's it. The user will do

`<Tabs ... />` for primary and `<Tabs secondary .../>` for secondary

Reply:
Cool! That's a lot better. We probably won't have more types and that's super easy to read. Thanks 😎